File: qt/gen_qtyperevision.cpp
// +build ignore

#include <QTypeRevision>
#include <qtyperevision.h>
#include "gen_qtyperevision.h"

#ifndef _Bool
#define _Bool bool
#endif

QTypeRevision* QTypeRevision_new() {
	return new QTypeRevision();
}

QTypeRevision* QTypeRevision_new2(QTypeRevision* param1) {
	return new QTypeRevision(*param1);
}

QTypeRevision* QTypeRevision_Zero() {
	return new QTypeRevision(QTypeRevision::zero());
}

bool QTypeRevision_HasMajorVersion(const QTypeRevision* self) {
	return self->hasMajorVersion();
}

unsigned char QTypeRevision_MajorVersion(const QTypeRevision* self) {
	quint8 _ret = self->majorVersion();
	return static_cast<unsigned char>(_ret);
}

bool QTypeRevision_HasMinorVersion(const QTypeRevision* self) {
	return self->hasMinorVersion();
}

unsigned char QTypeRevision_MinorVersion(const QTypeRevision* self) {
	quint8 _ret = self->minorVersion();
	return static_cast<unsigned char>(_ret);
}

bool QTypeRevision_IsValid(const QTypeRevision* self) {
	return self->isValid();
}

void QTypeRevision_Delete(QTypeRevision* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QTypeRevision*>( self );
	} else {
		delete self;
	}
}

File: qt/gen_qtyperevision.go
package qt

import (
	"unsafe"
)

type QTypeRevision struct {
	h          uintptr
	isSubclass bool
}

// NewQTypeRevision constructs a new QTypeRevision object.
func NewQTypeRevision() *QTypeRevision {

	ret := newQTypeRevision(QTypeRevision_new())
	ret.isSubclass = true
	return ret
}

// NewQTypeRevision2 constructs a new QTypeRevision object.
func NewQTypeRevision2(param1 *QTypeRevision) *QTypeRevision {

	ret := newQTypeRevision(QTypeRevision_new2(param1.cPointer()))
	ret.isSubclass = true
	return ret
}

func QTypeRevision_Zero() *QTypeRevision {
	_goptr := newQTypeRevision(QTypeRevision_Zero())
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QTypeRevision) HasMajorVersion() bool {
	return (bool)(QTypeRevision_HasMajorVersion(this.h))
}

func (this *QTypeRevision) MajorVersion() byte {
	return (byte)(QTypeRevision_MajorVersion(this.h))
}

func (this *QTypeRevision) HasMinorVersion() bool {
	return (bool)(QTypeRevision_HasMinorVersion(this.h))
}

func (this *QTypeRevision) MinorVersion() byte {
	return (byte)(QTypeRevision_MinorVersion(this.h))
}

func (this *QTypeRevision) IsValid() bool {
	return (bool)(QTypeRevision_IsValid(this.h))
}
